source/Halibut/Transport/Protocol/TemporaryFileStream.cs

```diff
@@ -39,15 +39,22 @@ public void SaveTo(string filePath)
 
         void SetFilePermissionsToInheritFromParent(string filePath)
         {
-            var fileInfo = new FileInfo(filePath);
+            try
+            {
+                var fileInfo = new FileInfo(filePath);
 #pragma warning disable PC001 // API not supported on all platforms
-            var fileSecurity = fileInfo.GetAccessControl();
+                var fileSecurity = fileInfo.GetAccessControl();
 
-            //When isProtected (first param) is false, SetAccessRuleProtection changes the permissions of the file to allow inherited permissions. 
-            //preserveInheritance (second param) is ignored when isProtected is false.
-            fileSecurity.SetAccessRuleProtection(false, false);
-            fileInfo.SetAccessControl(fileSecurity);
+                //When isProtected (first param) is false, SetAccessRuleProtection changes the permissions of the file to allow inherited permissions. 
+                //preserveInheritance (second param) is ignored when isProtected is false.
+                fileSecurity.SetAccessRuleProtection(false, false);
+                fileInfo.SetAccessControl(fileSecurity);
 #pragma warning restore PC001 // API not supported on all platforms
+            }
+            catch (UnauthorizedAccessException ex)
+            {
+                log.Write(EventType.Security, $"Ignoring an unauthorized access issue: {ex.Message}. {nameof(TemporaryFileStream)} assumes that filesystem permissions allow full control to the executing user for {filePath}. Update those permissions to remove this log.");
```

Comment:
Ooo, one thing .. can `log` be null? Should it need a null check `log?.Write...` ? Worth checking just in case

Reply:
Good pick @thebigmoosey hadn't considered that.